File: Сортировка_вставками.cpp
#include <iostream>
using namespace std;

void insertionSort(int array[], int size) {
    // Начинаем со второго элемента
    for (int i = 1; i < size; i++) {
        int key = array[i]; // Текущий элемент для вставки
        int j = i - 1;
        
        // Сдвигаем элементы большие key вправо
        while (j >= 0 && array[j] > key) {
            array[j + 1] = array[j];
            j--;
        }
        array[j + 1] = key; // Вставляем key на правильную позицию
    }
}

int main() {
    int array[] = {12, 11, 13, 5, 6};
    int size = sizeof(array) / sizeof(array[0]);
    
    cout << "Исходный массив: ";
    for (int i = 0; i < size; i++) {
        cout << array[i] << " ";
    }
    
    insertionSort(array, size);
    
    cout << "\nОтсортированный массив: ";
    for (int i = 0; i < size; i++) {
        cout << array[i] << " ";
    }
    return 0;
}

File: Сортировка_вставками.py
def insertion_sort(arr):
    # Начинаем со второго элемента
    for i in range(1, len(arr)):
        key = arr[i]  # Текущий элемент для вставки
        j = i - 1
        
        # Сдвигаем элементы большие key вправо
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key  # Вставляем key на правильную позицию

if __name__ == "__main__":
    array = [12, 11, 13, 5, 6]
    print("Исходный массив:", array)
    insertion_sort(array)
    print("Отсортированный массив:", array)
